File: API/Entities/Stock.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Stock
    {
        public int Id { get; set; }
        public int Quantity { get; set; }
        public long TotalPrice { get; set; }
        public int ProductId { get; set; }
        public Product Product { get; set; }
        public int SenderId { get; set; }
        public Sender Sender { get; set; }
        public int CategoryId{get;set;}
        public Category Category {get;set;}

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        
    }

}

File: my-app/src/app/models/Stock.ts
export interface Stock {
    id:         number;
    quantity:   number;
    totalPrice: number;
    productId:  number;
    product:    Product;
    senderId:   number;
    sender:     Sender;
    categoryId: number;
    category:   Category;
    orderDate:  Date;
}

export interface Category {
    id:    number;
    brand: string;
    type:  string;
}

export interface Product {
    id:              number;
    name:            string;
    imageUrl:        string;
    description:     string;
    price:           number;
    quantityInStock: number;
    categoryId:      number;
    category:        Category;
}

export interface Sender {
    id:         number;
    name:       string;
    tellNumber: number;
    address:    string;
}
